src/common/ad_tx.c

```diff
@@ -821,9 +927,50 @@ static umem_off_t
 umo_tx_alloc(struct umem_instance *umm, size_t size, int slab_id, uint64_t flags,
 	     unsigned int type_num)
 {
+	struct ad_tx		*tx = tx_get();
 	struct ad_blob_handle	 bh = umm2ad_blob_hdl(umm);
+	struct ad_blob		*blob = bh.bh_blob;
+	void			*ptr;
+	umem_off_t		 off;
+
+	off = ad_alloc(bh, 0, size, NULL);
+	if (!UMOFF_IS_NULL(off) && !(flags & UMEM_FLAG_NO_FLUSH)) {
```

Comment:
Is there any use case that does tx_alloc() with NO_FLUSH flag? I suppose tx_alloc() is only used for index (values are all allocated by reserve() or atomic_alloc()?).

Reply:
if using NO_FLUSH flag is incorrect in our usage model, we probably can assert it.

Reply:
Yes, I think returning an error or assert could help us to catch potential bugs.